content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md

```diff
@@ -0,0 +1,1224 @@
+<script>
+
+    ...
+
+    const editRole = document.querySelectorAll('.edit')
+    const deleteUser = document.querySelector('.delete')
+  editRole.forEach((button, i) => {
+    button.addEventListener('click', async() => {
+      display.textContent= ''
+      const id = data.user[i+1].id
+      const res = await fetch('/api/auth/update', {
+      method: 'PUT',
+      body: JSON.stringify({ role: 'admin', id}),
+      headers: { 'Content-Type': 'application/json' }
+      })
+      const dataUpdate = await res.json()
+      if (res.status === 400 || res.status === 401) {
+        document.body.scrollTop = 0
+        document.documentElement.scrollTop = 0
+        return display.textContent = `${dataUpdate.message}. ${dataUpdate.error ? dataUpdate.error : ''}`
+      }
+      location.assign('/admin')
+      })
+    });
+
+      ...
+</script>
+```
+
+### Delete Users
+
+Deleting Users from the database should be the duty of an admin.
+
+`admin.ejs`:
+
+```html
+<script>
+
+  ...
+
+  const editRole = document.querySelectorAll('.edit')
+  const deleteUser = document.querySelector('.delete')
+  deleteUser.forEach((button, i)=> {
+   button.addEventListener('click', async ()=> {
+   display.textContent =''
+   const id = data.user[i+1].id
+   const res = await fetch('/api/auth/deleteUser', {
+     method: 'DELETE',
+     body: JSON.stringify({id}),
+     headers: {'Content-Type': 'application/json'}
+     })
+   const dataDelete = await res.json()
+   if (res.status === 401){
+     document.body.scrollTop = 0
+     document.documentElement.scrollTop = 0
+     return display.textContent = `${dataUpdate.message}. ${dataUpdate.error ? dataUpdate.error : ''}`
+   }
+   location.assign('/admin')
+    })
+  })
+
+   ...
+</script>
+```
+
+You've created an event listener that listens for a click on the `Delete User` button. When the button is clicked, you'll send a `DELETE` request to `/api/auth/deleteUser`.
+
+> Please make sure the admin user is first on the list to avoid populating the database with an admin user again.
+
+## Logout Functionality
+
+To log out users, you need to remove the token from the client and redirect the client to the home page.
+
+You'll create a `GET` request to `/logout` in the `server.js` file:
+
+```js
+app.get("/logout", (req, res) => {
+  res.cookie("jwt", "", { maxAge: "1" })
+  res.redirect("/")
+})
+```
+
+The code snippet replaced the JWT token with an empty string and gave it a lifespan of 1 second.
+
+After creating the `GET` request, add a logout button to the admin's route and user's route:
+
+```html
+...
+<ul></ul>
+<button class="logout"><a href="/logout">Log Out</a></button>
+...
+```
+
+## Conclusion
+
+You've successfully learned how to perform CRUD operations using MongoDB as database, hash passwords, authenticate using JWT, and render Embedded JavaScript (EJS). You can put all these together to build more complex applications.
+
+## Resources
+
+- [GitHub Repo](https://github.com/uma-victor1/role_auth)
```

Comment:
@uma-victor1 Please add your sample code in the following repo by raising a separate pull request.
​​
​​https://github.com/LoginRadius/engineering-blog-samples

Reply:
Done.